### ISSUE TITLE
Skip to main content link navigates to different page

### ISSUE THREAD

**Issue by AlanGreene** [tektoncd/dashboard#2049]
# Describe the bug

Using the keyboard, tab from the browser's address bar to the page content and a 'Skip to main content' link should be displayed. Click this link and the browser is navigated to the PipelineRuns page.

![image](https://user-images.githubusercontent.com/2829095/119158122-ca0c8b80-ba4d-11eb-88fb-a38abaedb533.png)

# Expected behaviour

Clicking the link should focus the main content of the current page, bypassing the navigation.

# Steps to reproduce the bug

- navigate to any page on the Dashboard, e.g. https://dashboard.robocat.tekton.dev/#/about
- press the tab key until the 'Skip to main content' link appears
- press enter to activate the link
- see the page is redirected to the PipelineRuns list

# Environment details

- Versions:
   - Tekton Dashboard: `main`

# Additional Info

The target of the link is `#main-content` which references the id of the element wrapping the main page content. Normally this would be OK and clicking the link would move focus to the referenced element, but for the Dashboard we're using the fragment to identify the current page. Since `main-content` isn't a recognised page in the app, we end up redirecting to the default page which is PipelineRuns.

We need to find a different strategy for this skip link, some suggestions to consider:
- `element.scrollIntoView()`
- `element.focus()`
- …

We cannot switch from HashRouter (e.g. `/#/about`) to BrowserRouter (e.g. `/about`) as the latter will not work in all cases depending on how the Dashboard is accessed.
